### PR TITLE
PV/PVC metrics updates

### PR DESCRIPTION
**Pull request by ashish2207**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/73 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit Tests
- [x] Tested in Kubernetes cluster

Unit Test Results:
[csi-powerstore]# make test
go clean -cache; cd ./pkg; go test -race -cover -coverprofile=coverage.out -coverpkg ./... ./...
ok      github.com/dell/csi-powerstore/pkg/array        0.142s  coverage: 3.3% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/common       0.143s  coverage: 1.3% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/common/fs    0.170s  coverage: 1.0% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/controller   0.740s  coverage: 38.7% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/identity     0.144s  coverage: 0.2% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/interceptors 2.883s  coverage: 1.9% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/node 0.853s  coverage: 39.0% of statements in ./...
ok      github.com/dell/csi-powerstore/pkg/tracer       0.142s  coverage: 0.2% of statements in ./...

Code Coverage: 85.6%

### CODE REVIEW

**Review by harishp8889 — APPROVED** [review]
@ashish2207  LGTM, Please add unit test report

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -973,46 +973,46 @@ func (s *Service) ControllerGetVolume(ctx context.Context, req *csi.ControllerGe
...
+			nfsExport, err := s.Arrays()[arrayID].Client.GetNFSExportByFileSystemID(ctx, fs.ID)
+			if err != nil {
+				if apiError, ok := err.(gopowerstore.APIError); !ok || !apiError.NotFound() {
+					return nil, status.Errorf(codes.NotFound, "failed to find nfs export for filesystem with error: %v", err.Error())
+				}
+			} else {
+				// get hosts publish to export
+				hosts = append(nfsExport.ROHosts, nfsExport.RORootHosts...)
+				hosts = append(hosts, nfsExport.RWHosts...)
+				hosts = append(hosts, nfsExport.RWRootHosts...)
```
Comment (francis-nijay) [inline pkg/controller/controller.go:989]: Return abnormal if no hosts are found

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -973,46 +973,46 @@ func (s *Service) ControllerGetVolume(ctx context.Context, req *csi.ControllerGe
...
-			return nil, status.Errorf(codes.NotFound, "failed to find filesystem: %s with error: %v", id, err.Error())
-		}
-
-		// get exports for filesystem if exists
-		nfsExport, err := s.Arrays()[arrayID].Client.GetNFSExportByFileSystemID(ctx, fs.ID)
-		if err != nil {
-			if apiError, ok := err.(gopowerstore.APIError); !ok || !apiError.NotFound() {
-				return nil, status.Errorf(codes.NotFound, "failed to find nfs export for filesystem with error: %v", err.Error())
-			}
+			abnormal = true
```
Comment (francis-nijay) [inline pkg/controller/controller.go:976]: Return abnormal only for 404 - FS not found. Return error in case of any other status code

**Review by francis-nijay — CHANGES_REQUESTED** [review]

File: pkg/controller/controller.go (hunk 1 above)
Comment (ashish2207) [reply pkg/controller/controller.go:992]: done

File: pkg/controller/controller.go (hunk 2 above)
Comment (ashish2207) [reply pkg/controller/controller.go:979]: done

**Comment by ashish2207** [comment]
> @ashish2207 LGTM, Please add unit test report

done

**Review by harishp8889 — APPROVED** [review]

**Review by francis-nijay — APPROVED** [review]

**Pull request merged**